ocaml/typing/typedtree.ml

```diff
@@ -122,11 +122,10 @@ and expression =
    }
 
 and exp_extra =
-  | Texp_constraint of core_type
+  | Texp_constraint of core_type option * Parsetree.mode loc list
```

Comment:
The `core_type` here is `typedtree.core_type`, translated from `parsetree.core_type` via `transl_simple_type`.
Similarly, we should have `Mode.Alloc.Const.Option.t`, translated from `parsetree.mode loc list` via `transl_mode_annotes`.